securedrop_client/storage.py

```diff
@@ -940,6 +954,24 @@ def delete_single_submission_or_reply_on_disk(
             pass
 
 
+def _cleanup_directory_if_empty(target_dir: str) -> None:
+    """
+    If a directory is empty, remove it. Used after sync to ensure that an empty
+    directory is not left behind after files and messages are deleted.
+    """
+
+    try:
+        if os.path.isdir(target_dir):
+            if not next(os.scandir(target_dir), None):
```

Comment:
Very efficient to check `scandir()` rather than `len() == 0` on `os.listdir()` or `os.walk()`; nice!